data-loader/core/src/main/java/com/scalar/db/dataloader/core/dataimport/dao/ScalarDBDao.java

```diff
@@ -0,0 +1,431 @@
+        createScan(
+            namespace, tableName, partitionKey, scanRange, sortOrders, projectionColumns, limit);
+    try {
+      return storage.scan(scan);
+    } catch (ExecutionException e) {
+      throw new ScalarDBDaoException(CoreError.DATA_LOADER_ERROR_SCAN.buildMessage(), e);
+    }
+  }
+
+  /**
+   * Create ScalarDB scan instance
+   *
+   * @param namespace ScalarDB namespace
+   * @param tableName ScalarDB table name
+   * @param partitionKey Partition key used in ScalarDB scan
+   * @param scanRange Optional range to set ScalarDB scan start and end values
+   * @param sortOrders Optional scan clustering key sorting values
+   * @param projectionColumns List of column projection to use during scan
+   * @param limit Scan limit value
+   * @return ScalarDB scan instance
+   */
+  Scan createScan(
+      String namespace,
+      String tableName,
+      Key partitionKey,
+      ScanRange scanRange,
+      List<Scan.Ordering> sortOrders,
+      List<String> projectionColumns,
+      int limit) {
+    // If no partition key is provided a scan all is created
+    if (partitionKey == null) {
+      ScanBuilder.BuildableScanAll buildableScanAll =
+          Scan.newBuilder().namespace(namespace).table(tableName).all();
+
+      // projection columns
+      if (projectionColumns != null && !projectionColumns.isEmpty()) {
+        buildableScanAll.projections(projectionColumns);
+      }
+
+      // limit
+      if (limit > 0) {
+        buildableScanAll.limit(limit);
+      }
+      return buildableScanAll.build();
+    }
+
+    // Create a scan with partition key (not a scan all)
+    ScanBuilder.BuildableScan buildableScan =
+        Scan.newBuilder().namespace(namespace).table(tableName).partitionKey(partitionKey);
+
+    // Set the scan boundary
+    if (scanRange != null) {
+      // Set boundary start
+      if (scanRange.getScanStartKey() != null) {
+        buildableScan.start(scanRange.getScanStartKey(), scanRange.isStartInclusive());
+      }
+
+      // with end
+      if (scanRange.getScanEndKey() != null) {
+        buildableScan.end(scanRange.getScanEndKey(), scanRange.isEndInclusive());
+      }
+    }
+
+    // clustering order
+    for (Scan.Ordering sort : sortOrders) {
+      buildableScan.ordering(sort);
+    }
+
+    // projections
+    if (projectionColumns != null && !projectionColumns.isEmpty()) {
+      buildableScan.projections(projectionColumns);
+    }
+
+    // limit
+    if (limit > 0) {
+      buildableScan.limit(limit);
+    }
+    return buildableScan.build();
+  }
+
+  /**
+   * Return a ScalarDB get based on provided parameters
+   *
+   * @param namespace Namespace name
+   * @param tableName Table name
+   * @param partitionKey Partition key
+   * @param clusteringKey Optional clustering key for get
+   * @return ScalarDB Get instance
+   */
+  private Get createGetWith(
+      String namespace, String tableName, Key partitionKey, Key clusteringKey) {
+    if (clusteringKey != null) {
```

Comment:
[minor] I think we can reduce the duplicated code by setting Clustering Key like this as well as `createPutWith()`

```java
    Buildable buildable = ...

    if (clusteringKey != null) {
      buildable.clusteringKey(clusteringKey);
    }
```